components/brave_wallet/browser/ethereum_provider_impl.cc

```diff
@@ -314,7 +319,7 @@ void EthereumProviderImpl::OnGetNetworkAndDefaultKeyringInfo(
     // If the chain id is not known yet, then get it and set it first
     if (tx_data_1559->chain_id == "0x0" || tx_data_1559->chain_id.empty()) {
       json_rpc_service_->GetChainId(
-          mojom::CoinType::ETH,
+          mojom::CoinType::ETH, origin,
```

Comment:
Is this `if` condition still possible?
The chain param is from `json_rpc_service_->GetNetwork` which already calls `GetChainId` inside, we're just doing the same thing again here.
Seems like we can get rid of this if/else and just do GetAllowedAccounts.

Reply:
Yes, it is redundant and can be removed.

Reply:
fixed in 210bbec4829bea9ca6ae39e73d2bbd6bb233ad9e